pandas/core/series.py

```diff
@@ -1022,7 +1022,12 @@ def __getitem__(self, key):
         elif key_is_scalar:
             return self._get_value(key)
 
-        if is_hashable(key):
+        # Convert generator to list before going through hashable part
+        # (We will iterate through the generator there to check for slices)
+        if is_iterator(key):
+            key = list(key)
+
+        if is_hashable(key) and not isinstance(key, slice):
```

Comment:
very late review comment: would it make sense to make is_hashable_non_slice or something?  it wouldn't surprise me if many places that use is_hashable current assume non-slice but didn't get updated by this PR

Reply:
Yeah sure, I just updated the places that broke in the tests.

Is there a case where we would actually want ``is_hashable`` for a slice to equal True, though?
(I was thinking it might be cleaner to make ``is_hashable`` always return False for slices. This would be a breaking API change, though).

Reply:
i think having is_hashable be anything other than a try/except around hash would cause problems.  im suggesting a new function to de-duplicate the hashable-but-not-slice checks this introduces

Reply:
Made an issue #55152. 

I don't have the same time that I had in summer to work on pandas, but I'll try to have a look (no promises, though).